discussion-questions.md

```diff
@@ -1,8 +1,12 @@
 #  How much time did you spend on this assignment?
 
-_replace this text with your answer_
-
+I spent a total of approximately 5.5 hours on this assignment:
+- 2.5 hours on parts 1-4 of the assignment, installing and familiarizing myself with Scala
+- 2 hours on parts 5-6 of the assignment in building my own functions, adding fluency to the picture library, and modifying the tests accordingly
+- 1 hour on attempting to implement new functions, i.e. blur, bisectHorizontally, and bisectVertically. I did not have much trouble creating the bisecting functions, however, I spent a considerable amount of time trying to understand the algorithm for and attempting to comprehend which functions and methods to use to implement the Solarize and Blur functions.
 # Discussion questions / topics
 Below, add one or two things that you might want to discuss when we meet again.
 
-_replace this text with your discussion questions / topics_
\ No newline at end of file
+- In using Scala, I found many similarities between it and Python as well as Java. A question I wanted to ask was that why is it then that we use Scala, and not Python or Java, in implementing DSLs? Does it have some features which make it better-suited for this purpose?
+- I wasn't able to view the output image of my PictureProgram, I would really like to discuss a little more about the structure of a Scala program when using Image Processing
+- I would like to please discuss the algorithms for the Solarize and Blur function as well as the parameters needed to be able to implement these.
```

Comment:
Yes, solarize ended up being harder to implement than I intended!

Both solarize and blur look at a region of pixels, to determine the color of each individual pixel. These operations are examples of _convolution_ [1]. One tricky bit is: what happens when the region lies outside the image? For example, how do we blur all the pixels on the top edge of the image, when those pixels depend on the pixels in the non-existent row above them?

There are a few ways to deal with regions that fall outside the image. 
   1. Use 0 (i.e., black) or some other hard-coded color, which essentially crops the image.
  1. Extend the image, essentially copying the edges out one more row.
  1. Fold the region back into the image, essentially mirroring the image along the edges.
  1. Wrap the image around, so that you're reading from the bottom of the image, when the region extends past the top of the image.
All of these techniques require our code to determine when part of the region is outside the image.

[1] https://en.wikipedia.org/wiki/Kernel_(image_processing)